### PR TITLE
adding an ONNX demo

### PR DESCRIPTION
**Pull request by staylorTT**
Adding an ONNX resnet demo.

### CODE REVIEW

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+from PIL import Image
+from transformers import AutoFeatureExtractor, ResNetForImageClassification
+
+import forge
+from forge.verify.verify import verify
+from forge.verify.config import VerifyConfig
+from forge.verify.value_checkers import AutomaticValueChecker
+
+
+def run_resnet_onnx(variant="microsoft/resnet-50", batch_size=1, opset_version=11):
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:21]: Let's use the same opset as on FFE:
```
opset_version=17
```

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+    Args:
+        variant (str): The HuggingFace model variant to use
+        batch_size (int): The batch size for inference
+        opset_version (int): The ONNX opset version to use
+    """
+    print(f"Running ResNet ONNX demo with variant: {variant}, batch size: {batch_size}")
+
+    # Load ResNet feature extractor and model checkpoint from HuggingFace
+    model_ckpt = variant
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:33]: Redundant. We should just use variant in `from_pretrained ` funcitons

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    feature_extractor = AutoFeatureExtractor.from_pretrained(model_ckpt)
+    torch_model = ResNetForImageClassification.from_pretrained(model_ckpt)
+
+    # Create a temporary path for the ONNX model
+    onnx_path = "resnet50.onnx"
+
+    # Load data sample
+    url = "https://images.rawpixel.com/image_1300/cHJpdmF0ZS9sci9pbWFnZXMvd2Vic2l0ZS8yMDIyLTA1L3BkMTA2LTA0Ny1jaGltXzEuanBn.jpg"
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:41]: It'll be cleaner to use common libraries for small datasets instead of iterating over a single image. E.g.
```py
    from datasets import load_dataset

    dataset = load_dataset("zh-plus/tiny-imagenet")
    images = random.sample(dataset["valid"]["image"], 10)
```

Not a mandatory requirement as this demo uses only a single batch. But recommendation if we want to run multiple samples (or batches) even with single batch dim.

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    batch_input = torch.cat(pixel_values, dim=0)
+
+    # Export model to ONNX
+    print(f"Exporting model to ONNX with opset version {opset_version}...")
+    torch.onnx.export(
+        torch_model,
+        batch_input,
+        onnx_path,
+        opset_version=opset_version,
+        input_names=["input"],
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:57]: This and following 2 lines are redundant.

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        dynamic_axes={"input": {0: "batch_size"}, "output": {0: "batch_size"}},
+    )
+
+    # Load ONNX model
+    print("Loading ONNX model...")
+    onnx_model = onnx.load(onnx_path)
+    onnx.checker.check_model(onnx_model)
+
+    # Create a module name for the model
+    module_name = f"onnx_resnet50_{variant.split('/')[-1]}"
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:68]: We should just do this inline while wrapping the model. E.g.
```py
framework_model = forge.OnnxModule(f"onnx_resnet50_{variant.split('/')[-1]}", onnx_model)
```

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    predicted_value = output[0].argmax(-1)
+    predicted_label = [torch_model.config.id2label[pred.item()] for pred in predicted_value]
+
+    # Print results
+    print("\nResults:")
+    for sample in range(batch_size):
+        print(f"True Label: {label[sample]} | Predicted Label: {predicted_label[sample]}")
+
+    # Verify results
+    print("\nVerifying results...")
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:94]: There is not value of doing verifications (on tensor level) in demo scripts. We should remove these

**Comment by staylorTT** [comment]
I told windsurf to implement your feedback, let me know how it did.

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+    # Create a temporary path for the ONNX model
+    onnx_path = "resnet50.onnx"
+
+    # Load data samples from a dataset
+    print("Loading dataset...")
+    dataset = load_dataset("zh-plus/tiny-imagenet")
+
+    # Select random images for the batch
+    sample_indices = random.sample(range(len(dataset["valid"])), batch_size)
```
Comment (nvukobratTT) [inline demos/tt-forge-fe/cnn/resnet_onnx_demo.py:42]: I assume we would like to have repeatable examples. Therefore, I would advise you not to use random sampling here. Instead, let's pick up N fist samples and run inference on those

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py (hunk 4 above)
Comment (nvukobratTT) [reply demos/tt-forge-fe/cnn/resnet_onnx_demo.py:66]: Same comment, the bellow ones are redundant for this test case.

**Review by staylorTT — COMMENTED** [review]
OK, I think the feedback has been addressed now. @nvukobratTT

File: demos/tt-forge-fe/cnn/resnet_onnx_demo.py (hunk 4 above)
Comment (nvukobratTT) [reply demos/tt-forge-fe/cnn/resnet_onnx_demo.py:65]: nit: Inputs line is still there (others have been removed)

**Review by nvukobratTT — APPROVED** [review]
Added minor comment, but this change looks good
